### ISSUE TITLE
Creating a working PoC and tracking Progress Here tidb wasm nodejs

### ISSUE THREAD

**Issue by frank-dspeed** [pingcap/tidb#15520]
## Development Task
This will Protocol the Journey of running tidb wasm inside NodeJS 13+ with wasm support and will catch up any findings inspired by https://github.com/pingcap/tidb/issues/13236 tidb wasm + wasi on wasmer.

**Comment by siddontang** [comment]
PTAL @lucklove

**Comment by joe-getcouragenow** [comment]
Is running in the browser using tinygo still a target ?

**Comment by lucklove** [comment]
@joe-getcouragenow No. At this time we face another problem: the master branch can't compile with standard golang even. TiDB has too many symbols,  however, the wasm32's limit is less than 65535.

**Comment by lucklove** [comment]
@frank-dspeed I'm not familiar with NodeJS，can you help on this? Maybe we can compile tidb v3.x to NodeJS.

**Comment by frank-dspeed** [comment]
@lucklove at present we are facing a v8 internal issue there are some open issues in chromium with your current web build as it was able to Freeze the engine thats why i did not reported that back it is also considered a Security issue there are tons of engineers working on that.

tidb works great with wasmer at present. Wasmer is a universal runtime that is also use able embedded in nodejs and then v8 does not freeze as it is not even used :).

**Comment by joe-getcouragenow** [comment]
> @lucklove at present we are facing a v8 internal issue there are some open issues in chromium with your current web build as it was able to Freeze the engine thats why i did not reported that back it is also considered a Security issue there are tons of engineers working on that.
> 
> tidb works great with wasmer at present. Wasmer is a universal runtime that is also use able embedded in nodejs and then v8 does not freeze as it is not even used :).

@frank-dspeed Can you link to this ? been meaning to look into using WASMER for Server side ...

**Comment by frank-dspeed** [comment]
@joe-getcouragenow i did not write that but i need to point out lol you can even run wasmer on the client :) wasmer is amazing they did even compile nginx to wasm and runned it…

**Comment by joe-getcouragenow** [comment]
> @joe-getcouragenow i did not write that but i need to point out lol you can even run wasmer on the client :) wasmer is amazing they did even compile nginx to wasm and runned it…

@frank-dspeed  Can you link to anything showing this ? would like to play around with this very much.
